src/utils/api/helper.ts

```diff
@@ -629,6 +629,40 @@ export const proceedToPayment = async (
     O.getOrElseW(() => undefined)
   );
 
+  const details = pipe(
+    getSessionItem(SessionItems.paymentMethod) as PaymentMethod | undefined,
+    O.fromNullable,
+    O.map((method) => method.paymentTypeCode),
+    O.getOrElse(() => {
+      // eslint-disable-next-line no-console
+      console.error(
+        "Missing expected `paymentMethod.paymentTypeCode` in session storage!"
+      );
+      return "";
+    }),
+    (paymentTypeCode) => {
+      switch (paymentTypeCode) {
+        case "CP":
+          return {
+            detailType: "cards",
+            orderId:
+              (getSessionItem(SessionItems.orderId) as string | undefined) ||
+              "",
+          };
+        case "MYBK":
```

Comment:
maybe here we can work on `methodManagement` field that has a restricted domain.

- `REDIRECT` -> will be always a redirect auth request
- `NOT_ONBOARDABLE` -> will be always an apm auth request
- `ONBOARDABLE` -> here we need to discriminate between CP (cards) and others (only paypal by the moment) to send a cards/apm auth request.

This mapping could make this check valid also for future added payment methods.
What do you think?

Reply:
Postponing in a future PR as session storage `paymentMethod` is populated with the response from `/payment-methods/{id}/sessions/{orderId}`, which doesn't contain `methodManagement`